core/src/test/scala/org/apache/spark/util/EventLoopSuite.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+import org.scalatest.concurrent.Eventually._
+import org.scalatest.FunSuite
+
+class EventLoopSuite extends FunSuite {
```

Comment:
Could you expand the tests for this class to include things like:
- Throwing an error within onError()
- Stopping the event loop if the onReceive() is inside something like a Lock#acquireUninterruptibly()
- Post events from different threads and make sure nothing throws a ConcurrentModificationException :)

Currently the implementation is simple, but I'd like to make sure that future changes don't break some property of event loops which the DAGScheduler rarely or never exercises.